### ISSUE TITLE
Error with latest git-annex version and initremote --sameas

### ISSUE THREAD

**Issue by eds-slim** [datalad/datalad#3816]
In my dataset,I set up two special rsync remotes using `git-annex initremote NAS type=rsync ...` and `git-annex initremote NASuser --sameas NAS type=rsync .. autoenable=true.`, pointing to the same data source, but having different file permissions (`NASuser`is read-only).

Now, when I publish the dataset and install it somewhere else, I receive the error
```
[INFO   ]   Remote origin not usable by git-annex; setting annex-ignore 
[ERROR  ] sequence item 0: expected str instance, NoneType found [utils.py:_handle_possible_annex_dataset:192] (TypeError) 
```
Also, the special remote `NASuser`is not enabled automatically, and indeed, enabling it via `datalad siblings enable -s NASuser` is not successful, giving
```
[ERROR  ] unknown sibling name [query-sibling(/home/eckhard/Documents/Research/DataLad/testdemo)] 
```
Using `git-annex enableremote NASuser` (or indeed `git-annex init`) instead seems to work, resulting in
```
enableremote NASuser ok
(recording state in git...)
```
So, while the problem can be easily ignored for individual datasets, it does prevent me from doing recursive installs.

Is the `--sameas` option of `git-annex initremote` known to be problematic and are there plans to make `datalad`compatible with it?
Thanks!

**Comment by eds-slim** [comment]
Thanks @bpoldrack. I'm using `datalad --version` `datalad 0.11.7`, installed via `pip`.

**Comment by bpoldrack** [comment]
There seems to be no difference in `siblings` implementation to 012.0rc6.
However, when I try to replicate I fail so far. While `datalad install` doesn't report to enable `NASuser`, all looks actually fine:

```
/tmp/test-special-remote $ git annex initremote NAS type=rsync encryption=none rsyncurl=somefake
/tmp/test-special-remote $ git annex initremote NASuser --sameas NAS type=rsync encryption=none rsyncurl=somefake autoenable=true
/tmp/test-special-remote $ cd ..
/tmp $ datalad install -s test-special-remote/ clone
/tmp $ cd clone
/tmp/clone $ cat .git/config
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = /tmp/test-special-remote/
        fetch = +refs/heads/*:refs/remotes/origin/*
        annex-uuid = 89404499-ee32-4620-95f5-e7ebf2c56e06
[branch "master"]
        remote = origin
        merge = refs/heads/master
[annex]
        uuid = 9f6dbee2-e9e2-4b02-b4bd-3983458acf7b
        version = 5
[remote "NASuser"]
        annex-rsyncurl = somefake
        annex-uuid = 2b55d27d-8ef0-47c6-b5e4-d8070aef71d9
        annex-ignore = false
/tmp/clone $ git annex info
repository mode: indirect
trusted repositories: 0
semitrusted repositories: 6
        00000000-0000-0000-0000-000000000001 -- web
        00000000-0000-0000-0000-000000000002 -- bittorrent
        2b55d27d-8ef0-47c6-b5e4-d8070aef71d9 -- [NASuser]
        52680a92-573a-4cb1-91f7-a811299b63bd -- NAS
        89404499-ee32-4620-95f5-e7ebf2c56e06 -- ben@tree:/tmp/test-special-remote [origin]
        9f6dbee2-e9e2-4b02-b4bd-3983458acf7b -- ben@tree:/tmp/clone [here]
untrusted repositories: 0
transfers in progress: none
available local disk space: 1.97 gigabytes (+1 megabyte reserved)
local annex keys: 0
local annex size: 0 bytes
annexed files in working tree: 1
size of annexed files in working tree: 5 bytes
bloom filter size: 32 mebibytes (0% full)
backend usage: 
        MD5E: 1

/tmp/clone $ datalad siblings enable -s NASuser
.: NASuser(?) [git]
```
So, two questions, @eds-slim :

1. Could you try to install datalad 0.12.0rc6 and double-check whether it's still happening for you?
If it seems to work, please make sure to still try `datalad siblings enable -s NASuser`, since its error message you posted could be an issue  in its own right.
2. How did you determine it didn't work in the first place?

**Comment by eds-slim** [comment]
HI @bpoldrack thanks for your reply.I have yet to install `datalad version 0.12`, but let me already report what happens when I run your minimal example in `version 0.11.6`
```
MinimalExample$ datalad create new
[INFO   ] Creating a new annex repo at MinimalExample/new 
create(ok): MinimalExample/new (dataset)
MinimalExample$ cd new
MinimalExample/new$ /home/user/.local/bin//git-annex initremote NAS type=rsync rsyncurl=somefake encryption=none

initremote NAS ok
(recording state in git...)
MinimalExample/new$ /home/user/.local/bin//git-annex initremote NASuser --sameas NAS type=rsync rsyncurl=somefake encryption=none
initremote NASuser ok
(recording state in git...)
MinimalExample/new$ cd ..
MinimalExample$ datalad install -s new clone/
[INFO   ] Cloning new into 'MinimalExample/clone' 
[ERROR  ] sequence item 1: expected str instance, NoneType found [utils.py:_handle_possible_annex_dataset:203] (TypeError) 
MinimalExample$ cd clone/
MinimalExample/clone$ cat .git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = MinimalExample/new
	fetch = +refs/heads/*:refs/remotes/origin/*
	annex-uuid = 816ae088-77ce-449f-8b0a-656971af5ced
[branch "master"]
	remote = origin
	merge = refs/heads/master
[annex]
	uuid = b18c0f95-3182-4ab6-a425-1ddc947a207b
	version = 5
MinimalExample/clone$ /home/user/.local/bin//git-annex info

trusted repositories: 0
semitrusted repositories: 5
	00000000-0000-0000-0000-000000000001 -- web
 	00000000-0000-0000-0000-000000000002 -- bittorrent
 	816ae088-77ce-449f-8b0a-656971af5ced -- MinimalExample/new [origin]
 	89895e8f-ec58-47a6-8bc3-9de8c4503a4e -- NAS
 	b18c0f95-3182-4ab6-a425-1ddc947a207b -- MinimalExample/clone [here]
untrusted repositories: 0
transfers in progress: none
available local disk space: 1.27 gigabytes (+1 megabyte reserved)
local annex keys: 0
local annex size: 0 bytes
annexed files in working tree: 0
size of annexed files in working tree: 0 bytes
bloom filter size: 32 mebibytes (0% full)
backend usage: 
MinimalExample/clone$ datalad siblings enable -s NASuser
[ERROR  ] 'name' [siblings.py:<dictcomp>:744] (KeyError) 
MinimalExample/clone$ /home/user/.local/bin//git-annex enableremote NASuser
enableremote NASuser ok
(recording state in git...)
MinimalExample/clone$ /home/user/.local/bin//git-annex info
trusted repositories: 0
semitrusted repositories: 5
	00000000-0000-0000-0000-000000000001 -- web
 	00000000-0000-0000-0000-000000000002 -- bittorrent
 	816ae088-77ce-449f-8b0a-656971af5ced -- MinimalExample/new [origin]
 	89895e8f-ec58-47a6-8bc3-9de8c4503a4e -- NAS [NASuser]
 	b18c0f95-3182-4ab6-a425-1ddc947a207b -- MinimalExample/clone [here]
untrusted repositories: 0
transfers in progress: none
available local disk space: 1.27 gigabytes (+1 megabyte reserved)
local annex keys: 0
local annex size: 0 bytes
annexed files in working tree: 0
size of annexed files in working tree: 0 bytes
bloom filter size: 32 mebibytes (0% full)
backend usage: 
```

```
MinimalExample/clone$ datalad --version
datalad 0.11.7
MinimalExample/clone$ /home/user/.local/bin//git-annex version
git-annex version: 7.20191011-gd533e59e7
build flags: Assistant Webapp Pairing S3 WebDAV Inotify TorrentParser Feeds Testsuite
dependency versions: aws-0.21.1 bloomfilter-2.0.1.0 cryptonite-0.25 DAV-1.3.3 feed-1.0.1.0 ghc-8.6.5 http-client-0.5.14 persistent-sqlite-2.9.3 torrent-10000.1.1 uuid-1.3.13 yesod-1.6.0
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384 SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_224 SHA3_384E SHA3_384 SKEIN256E SKEIN256 SKEIN512E SKEIN512 BLAKE2B256E BLAKE2B256 BLAKE2B512E BLAKE2B512 BLAKE2B160E BLAKE2B160 BLAKE2B224E BLAKE2B224 BLAKE2B384E BLAKE2B384 BLAKE2BP512E BLAKE2BP512 BLAKE2S256E BLAKE2S256 BLAKE2S160E BLAKE2S160 BLAKE2S224E BLAKE2S224 BLAKE2SP256E BLAKE2SP256 BLAKE2SP224E BLAKE2SP224 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav adb tahoe glacier ddar git-lfs hook external
operating system: linux x86_64
supported repository versions: 7
upgrade supported from repository versions: 0 1 2 3 4 5 6
local repository version: 7
```

It is curious that `git-annex info` does not even list the `NASuser` repo, yet is able to enable it.

Also, in your output, while both `NAS` and `NASuser` are listed, they seem to have different UUIDs, which makes me wonder, if they really point to the same datasource. The `--sameas` option has only been available since `version 7.20191017` and, I believe, was silently igonred before, so the git-annex version might be worth doublechecking.

I will now upgrade datalad to 0.12 and report back.

**Comment by bpoldrack** [comment]
>   Also, in your output, while both NAS and NASuser are listed, they seem to have different UUIDs, which makes me wonder if they really point to the same datas ource. The --sameas option has only been available since version 7.20191017 and, I believe, was silently ignored before, so the git-annex version might be worth double-checking.

True. I missed that. 
However, I'll also double-check what's going on in 0.11.7's `install`. This line
> [ERROR  ] sequence item 1: expected str instance, NoneType found [utils.py:_handle_possible_annex_dataset:203] (TypeError) 

seems to be the actual issue. If that function raises, `install` might indeed not call `git annex init` and since this is, what is supposed to deal with autoenabling, it could be the reason for the failure. I'm not sure whether this is different in most recent datalad - I'll figure that out.

**Comment by kyleam** [comment]
This issue looks like it's shared by both 0.11.x and master.  The `install/clone` issue should be fixed by gh-3856, but as noted in that PR `publish` may need an update to properly support `sameas` remotes.
